components/signup/signup_email/signup_email.jsx

```diff
@@ -60,12 +60,24 @@ export default class SignupEmail extends React.Component {
     componentDidMount() {
         trackEvent('signup', 'signup_user_01_welcome');
 
+        this.setDocumentTitle(this.props.siteName);
+
         const {inviteId} = this.state;
         if (inviteId && inviteId.length > 0) {
             this.getInviteInfo(inviteId);
         }
     }
 
+    componentDidUpdate() {
+        this.setDocumentTitle(this.props.siteName);
+    }
+
+    setDocumentTitle = (siteName) => {
+        if (siteName) {
+            document.title = siteName;
+        }
+    }
```

Comment:
This is failing on signup_spec since the document title is hardcoded to "Mattermost".  Instead of removing "document title" check on `/signup` page, apply here the same implementation on `/login` page.